### PR TITLE
fixed some issues with JsonStorage (commit thread not terminated on s…

### PR DESCRIPTION
**Pull request by sobimibos**
fixed some issues with JsonStorage (commit thread not terminated on stop, timeout treated as nanos not as millis)

Signed-off-by: Michael Bock <mbock@gmxpro.de>

### CODE REVIEW

File: bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java [hunk 1]
```diff
@@ -295,49 +295,64 @@ public void commitDatabase() {
...
-
         // Create the timer task
         commitTimerTask = new CommitTimerTask();
 
         // Start the timer
         commitTimer.schedule(commitTimerTask, writeDelay);
     }
-
+    
+    public synchronized void deactivate() {
```
Comment (kaikreuzer) [inline bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java]: `deactivate` makes the impression as if there's a lifecycle management involved (and one would expect an `activate`method likewise).
As the storage is in general still usable after calling this method, would `flush` maybe be a better name?

Comment (kaikreuzer) [reply bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java]: Thinking about it, the `commitDatabase` method actually already IS a flush...

File: bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java [hunk 2]
```diff
@@ -295,49 +295,64 @@ public void commitDatabase() {
...
+        commitDatabase();
+        // Stop any existing timer
```
Comment (kaikreuzer) [inline bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java]: Shouldn't an existing timer be always cleared within the the `commitDatabase()` method?  If this were the case, we would not need this additional method at all.

File: bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java (hunk 1 above)
Comment (sobimibos) [reply bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java]: Can call it `stop`

Comment (kaikreuzer) [reply bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/internal/JsonStorage.java]: Did you read my other comment?

**Pull request merged**